2-WebApp-graph-user/2-5-HybridFlow/Controllers/LogoutController.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Identity.Web;
+using WebApp_OpenIDConnect_DotNet.Services;
+
+namespace WebApp_OpenIDConnect_DotNet.Controllers
+{
+    [Authorize]
+    [Route("[controller]")]
+    public class LogoutController : ControllerBase
+    {
+        private readonly IConfidentialClientApplicationService _confidentialClientApplicationService;
+        public LogoutController(IConfidentialClientApplicationService confidentialClientApplicationService)
+        {
+            _confidentialClientApplicationService = confidentialClientApplicationService;
+        }
+
+        [HttpPost]
+        [Authorize]
+        public async Task<IActionResult> PostAsync()
+        {
+            await _confidentialClientApplicationService.RemoveAccount(User.GetMsalAccountId());
+            await HttpContext.SignOutAsync("Cookies");
+
+            return Ok();
+        }
+    }
```

Comment:
Vanilla logout method here.

I make a call to the ConfidentialClientService to empty out the memory cache and clear the cookies used by WebApp that allows users access to the razor pages.

The authentication flow only tests if the user 'isAuthenticated'. Does not seem to be immediately germane to the access tokens the server might use.